### PR TITLE
Bug fixes

### PR DESCRIPTION
**Pull request by Pessimistress**
- Some stdlib functions are missing types
- Dependencies are not added when using self-assignment (e.g. `+=`)
- Test cases

@dy

### CODE REVIEW

File: lib/stdlib.js [hunk 1]
```diff
@@ -7,6 +7,18 @@
 
 var operators = require('./operators').operators;
 
+// Returns the type of argument at index
+// If argIndex is not specified, assume the first argument
+function genType(argIndex) {
```
Comment (dy) [inline lib/stdlib.js]: <3

**Comment by dy** [comment]
@Pessimistress I like the `genType` solution, follows the spec and makes less adhoc. Good eye!

**Comment by Pessimistress** [comment]
Can you publish a patch with this? :)

**Comment by dy** [comment]
@Pessimistress I've added you as an npm owner. You can publish the package too if you need that urgently.

**Pull request merged**